sotodlib/site_pipeline/make_atomic_filterbin_map.py

```diff
@@ -226,7 +216,41 @@ def future_write_to_log(e, errlog):
     f.close()
 
 
-def main(config_file=None, defaults=defaults, **args):
+def main(
+    context: str,
+    preprocess_config: str,
+    area: Optional[str] = None,
+    nside: Optional[int] = None,
+    query: Optional[str] = "type == 'obs' and subtype == 'cmb'",
+    odir: Optional[str] = "./output",
+    update_delay: Optional[float] = None,
+    site: Optional[str] = 'so_sat3',
+    hk_data_path: Optional[str] = None,
+    nproc: Optional[int] = 1,
+    atomic_db: Optional[str] = None,
+    comps: Optional[str] = 'TQU',
+    singlestream: Optional[bool] = False,
+    only_hits: Optional[bool] = False,
+    det_in_out: Optional[bool] = False,
+    det_left_right: Optional[bool] = False,
+    det_upper_lower: Optional[bool] = False,
+    scan_left_right: Optional[bool] = False,
+    ntod: Optional[int] = None,
+    tods: Optional[str] = None,
+    nset: Optional[int] = None,
+    wafer: Optional[str] = None,
+    freq: Optional[str] = None,
+    dec_ref: Optional[float] = -40.0,
+    center_at: Optional[str] = None,
+    max_dets: Optional[int] = None,
+    fixed_time: Optional[int] = None,
+    min_dur: Optional[int] = None,
+    verbose: Optional[int] = 0,
+    quiet: Optional[int] = 0,
+    window: Optional[float] = None,
+    dtype_tod: Optional[str] = 'float32',
+    dtype_map: Optional[str] = 'float64'):
```

Comment:
Carlos just an FYI, `Optional` is usually used when the parameter can be `None`. If it is not passed and it gets a non-None value then the optional is not needed.